CONTRIBUTIONS.md

```diff
@@ -0,0 +1,135 @@
+# Contribution Guide
+
+## Contributors
+
+Pull requests, issues and comments are welcomed. For pull requests:
+
+* Add tests for new features and bug fixes
+* Follow the existing style
+* Separate unrelated changes into multiple pull requests
+
+See the existing issues for things to start contributing.
+
+For bigger changes, make sure you start a discussion first by creating an issue and explaining the intended change.
+
+JFrog requires contributors to sign a Contributor License Agreement, known as a CLA. This serves as a record stating that the contributor is entitled to contribute the code/documentation/translation to the project and is willing to have it used in distributions and derivative works (or is willing to transfer ownership).
+
+
+## Build the Provider
+
+Simply run `make install` - this will compile the provider and install it to `~/terraform.d`. When running this, it will take the current tag and bump it 1 minor version. It does not actually create a new tag (that is `make release`). If you wish to use the locally installed provider, make sure your TF script refers to the new version number.
+
+Requirements:
+- [Terraform](https://www.terraform.io/downloads.html) 0.13
+- [Go](https://golang.org/doc/install) 1.15+ (to build the provider plugin)
+
+### Building on macOS
+
+This provider uses [GNU sed](https://www.gnu.org/software/sed/) as part of the build toolchain, in both Linux and macOS. This provides consistency across OSes.
+
+If you are building this on macOS, you have two options:
+- Install [gnu-sed using brew](https://formulae.brew.sh/formula/gnu-sed), OR
+- Use a Linux Docker image/container
+
+#### Using gnu-sed
+
+After installing with brew, get the GNU sed information:
+
+```sh
+$ brew info gnu-sed
+```
+
+You should see something like:
+```
+GNU "sed" has been installed as "gsed".
+If you need to use it as "sed", you can add a "gnubin" directory
+to your PATH from your bashrc like:
+
+     PATH="$(brew --prefix)/opt/gnu-sed/libexec/gnubin:$PATH"
+```
+
+Add the `gnubin` directory to your `.bashrc` or `.zshrc` per instruction so that `sed` command uses gnu-sed.
+
+## Testing
+
+How to run the tests isn't obvious.
+
+First, you need a running instance of the JFrog platform (RT and XR). However, there is no currently supported Dockerized, local version. You can ask for an instance to test against in as part of your PR or by messaging the maintainer in gitter.
```

Comment:
Users can test projects provider without Xray installed. So, running RT in a container should be enough. but we need to mention that no Xray functionality can be tested like this.

Reply:
Check the new file, I added a line about how the platform will help test project with artifactory and xray, and then the next line talks about the local version.